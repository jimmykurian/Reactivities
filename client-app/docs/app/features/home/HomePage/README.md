[**Client App Documentation v0.0.0**](../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../README.md) / app/features/home/HomePage

# app/features/home/HomePage

## Index

### Functions

- [default](functions/default.md)
