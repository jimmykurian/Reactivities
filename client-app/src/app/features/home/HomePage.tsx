/**
 * @file HomePage.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the HomePage component which serves as the home page of the application.
 */

import { Link } from 'react-router-dom';
import { Container, Header, Segment, Image, Button } from 'semantic-ui-react';

/**
 * @component HomePage
 * @description The HomePage component serves as the home page of the application. It displays a heading, a welcome message, and a button that links to the activities page within a container.
 *
 * @returns {JSX.Element} The rendered HomePage component.
 *
 * @remarks
 * This component uses the Semantic UI React library to create a styled container.
 * The container includes a heading, a welcome message, and a button that navigates to the activities page.
 * The component is styled with an inverted segment and centered text.
 *
 * @example
 * Here is an example of how to use the HomePage component:
 * ```tsx
 * import React from 'react';
 * import HomePage from './HomePage';
 *
 * const App = () => (
 *   <div>
 *     <HomePage />
 *   </div>
 * );
 *
 * export default App;
 * ```
 */
export default function HomePage(): JSX.Element {
  return (
    <Segment inverted textAlign="center" vertical className="masthead">
      <Container text>
        <Header as="h1" inverted>
          <Image
            size="massive"
            src="/assets/logo.png"
            alt="logo"
            style={{ marginBottom: 12 }}
          />
          Reactivities
        </Header>
        <Header as="h2" inverted content="Welcome to Reactivities" />
        <Button as={Link} to="/activities" size="huge" inverted>
          Take me to the Activities!
        </Button>
      </Container>
    </Segment>
  );
}
