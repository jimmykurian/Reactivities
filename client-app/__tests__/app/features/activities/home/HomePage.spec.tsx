import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import HomePage from '../../../../../src/app/features/home/HomePage';

// Mocking semantic-ui-react's Container component
jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
  Container: jest.fn(({ children, style }) => (
    <div style={style}>{children}</div>
  )),
}));

describe('HomePage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders the HomePage component with heading', () => {
    // Arrange & Act
    render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
    );

    // Assert
    const headingElement = screen.getByText(/Home page/i);
    expect(headingElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    // Arrange & Act
    const { asFragment } = render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
