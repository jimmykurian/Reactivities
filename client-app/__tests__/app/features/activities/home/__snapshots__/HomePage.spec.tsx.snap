// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="margin-top: 7em;"
  >
    <h1>
      Home page
    </h1>
  </div>
</DocumentFragment>
`;
